addons/resource/views/resource_views.xml

```diff
@@ -152,7 +152,7 @@
         <field name="priority">1</field>
         <field name="arch" type="xml">
             <tree string="Leave Detail">
-                <field name="name" string="Reason"/>
+                <field name="name" string="Reason" required="1"/>
```

Comment:
I think it would be better to make it required in the inherited view in `hr_holidays/views/rescource_views.xml` `resource_calendar_leaves_tree_inherit` rather than modifying the original behavior, by using 
```xml
<xpath expr="//field[@name='name']" position="attributes">
    <attribute name="string">Name</attribute>
    <attribute name="required">1</attribute>
</xpath>
```